apps/shipments/views.py

```diff
@@ -206,3 +207,22 @@ def create(self, request, *args, **kwargs):
         permission_link = PermissionLink.objects.create(**serializer.validated_data)
 
         return Response(PermissionLinkSerializer(permission_link).data, status=status.HTTP_201_CREATED)
+
+
+class ShipmentHistoryListView(viewsets.ViewSet):
+    http_method_names = ['get', ]
+    permission_classes = ((IsOwnerOrShared,) if settings.PROFILES_ENABLED else (permissions.AllowAny,))
+    pagination_class = ShipmentHistoryPagination
+    renderer_classes = (renderers.JSONRenderer, )
+
+    def list(self, request, *args, **kwargs):
+        shipment = Shipment.objects.get(id=kwargs['shipment_pk'])
+        queryset = shipment.history.all()
+        serializer = ChangesDiffSerializer(queryset)
+
+        paginator = self.pagination_class()
+        page = paginator.paginate_queryset(serializer.data, self.request, view=self)
+        if page is not None:
+            return paginator.get_paginated_response(page)
+
+        return Response(serializer.data)
```

Comment:
In which cases is this line hit?

Reply:
This is actually nerve hit, I just wanted to keep it consistent with drf code base.

Reply:
:+1: